alert-common/src/main/java/com/synopsys/integration/alert/common/event/BrokerServiceDependentTask.java

```diff
@@ -22,17 +22,18 @@
 import com.synopsys.integration.log.IntLogger;
 import com.synopsys.integration.log.Slf4jIntLogger;
 import com.synopsys.integration.wait.WaitJob;
-import com.synopsys.integration.wait.WaitJobTask;
 
 public class BrokerServiceDependentTask {
     private Logger logger = LoggerFactory.getLogger(getClass());
 
     private String taskToExecuteName;
     private Consumer<BrokerService> taskToExecuteOnceFound;
+    private BrokerServiceWaitTask brokerServiceWaitTask;
 
-    public BrokerServiceDependentTask(String taskToExecuteName, Consumer<BrokerService> taskToExecuteOnceFound) {
+    public BrokerServiceDependentTask(String taskToExecuteName, BrokerServiceWaitTask brokerServiceWaitTask, Consumer<BrokerService> taskToExecuteOnceFound) {
         this.taskToExecuteName = taskToExecuteName;
         this.taskToExecuteOnceFound = taskToExecuteOnceFound;
+        this.brokerServiceWaitTask = brokerServiceWaitTask;
     }
 
     public String getTaskToExecuteName() {
```

Comment:
This should have a dependency on the ExecutorService so that it executes in the same thread if needed for tests.